pandas/tests/frame/test_replace.py

```diff
@@ -1149,3 +1149,20 @@ def test_replace_method(self, to_replace, method, expected):
         result = df.replace(to_replace=to_replace, value=None, method=method)
         expected = DataFrame(expected)
         assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("replace_dict, final_data", [
+        ({'a': 1, 'b': 1}, [[3, 3], [2, 2]]),
+        ({'a': 1, 'b': 2}, [[3, 1], [2, 3]])
+    ])
+    def test_categorical_replace_with_dict(self, replace_dict, final_data):
+        # GH 26988
+        df = DataFrame([[1, 1], [2, 2]], columns=['a', 'b'], dtype='category')
+        expected = DataFrame(final_data, columns=['a', 'b'], dtype='category')
+        expected['a'].cat.set_categories([1, 2, 3], inplace=True)
+        expected['b'].cat.set_categories([1, 2, 3], inplace=True)
+        result = df.replace(replace_dict, 3)
+        assert_frame_equal(result, expected)
+        with pytest.raises(AssertionError):
```

Comment:
can you add a comment on what the error condition is, and why would it be an AssertionError?

Reply:
fixed

Reply:
wait, this is *really* odd, why is an AssertionError here?